src/equilibrate_utils.jl

```diff
@@ -0,0 +1,726 @@
+    p1 = copy(p); p1[rn.params_to_ints[param]] = range[1];
+    p2 = copy(p); p2[rn.params_to_ints[param]] = range[2];
+    sol1 = hc_solve_at(rn, p1)
+    sol2 = hc_solve_at(rn, p2)
+    d_sol = min(d_sol,sort(norm.([sol1...,sol2...]))[2]/10.)
+    tracker1 = make_coretracker(rn,sol1,p1,p2,dp/(range[2]-range[1]))
+    tracker2 = make_coretracker(rn,sol2,p2,p1,dp/(range[2]-range[1]))
+    paths_complete   = Vector{NamedTuple{(:p, :x),Tuple{Vector{Float64},Vector{Vector{Complex{Float64}}}}}}()
+    paths_incomplete = Vector{NamedTuple{(:p, :x),Tuple{Vector{Float64},Vector{Vector{Complex{Float64}}}}}}()
+    for sol in sol1
+        path = track_path(sol,tracker1,range...)
+        if (tracker1.state.status == CoreTrackerStatus.success)
+            remove_sol!(sol2,path.x[end],d_sol)
+            push!(paths_complete,path)
+        else
+            push!(paths_incomplete,path)
+        end
+    end
+    for sol in sol2
+        path = track_path(sol,tracker2,reverse(range)...)
+        (tracker2.state.status == CoreTrackerStatus.success) && remove_path!(paths_incomplete,path.x[end],d_sol)
+        push!(paths_complete,path)
+    end
+    append!(paths_complete,paths_incomplete)
+    using_temp_poly && finalise_solver!(rn)
+    return bifurcation_paths(positive_real_projection.(paths_complete),param,range[1],range[2],rn,p)
+end
+#Used in the homotopy continuation bifurcation traces heuristics. If the path was traced succesfully to its end, checks if it corresponds to a solution in that end and removes that solution (it does not need to be attempted in the other direction).
+function remove_sol!(results::Vector{Vector{ComplexF64}},path_fin::Vector{ComplexF64},d_sol::Float64)
+    for i = length(results):-1:1
+        if maximum(abs.([imag.(path_fin.-results[i])..., real.(path_fin.-results[i])...]))<d_sol
+            deleteat!(results,i)
+            return
+        end
+    end
+end
+#Used in the homotopy continuation bifurcation traces heuristics. Similar to the previous but removes an unfinished path.
+function remove_path!(paths::Vector{NamedTuple{(:p, :x),Tuple{Vector{Float64},Vector{Vector{Complex{Float64}}}}}},path_fin::Vector{ComplexF64},d_sol::Float64)
+    for i = length(paths):-1:1
+        if maximum(abs.([imag.(path_fin.-paths[i][2][1])..., real.(path_fin.-paths[i][2][1])...]))<d_sol
+            deleteat!(paths,i)
+            return
+        end
+    end
+end
+
+#--- Function for the main HC bifurcation solver ---#
+#Main bifurcation solver. Track paths from the first to the last parameter values. Detects whenever a solution is lost due to a complicated bifurcation. In this case it resumes the path tracking just after that bifurcation.
+#Δp = how long step in parameter value to take after a bifrucation is encountered (should be smaller than the minimum expected distance between two bifurcations).
+#Δx = distance between two solutions for them to be considered identical. Should larger than the distance between two solutions.
```

Comment:
if you mean `u`, then use `u`. We shouldn't have one file in DiffEq use a different naming convention from all of the others.

Reply:
Yes, I've went over the files to ensure there is not more x used.